File: Easy/InsertionSort.py
def insertionSort(array):
    for i in range(1,len(array)):
        j = i
        while j > 0 and array[j] < array[j-1]:
            swap(j, j-1, array)
            j -= 1

    return array

def swap(i, j, array):
    array[i],array[j]=array[j],array[i]

print(insertionSort([4,2,7,5,8,5,3,6,9]))

File: Easy/InsertionSort.cpp
#include<iostream>
#include<vector>
using namespace std;

void insertionSort(vector<int> &arr, int len)
{
    int j = 0;
    for(int i=0; i<len; i++)
    {
        j = i;
        while(j>0 && arr[j]<arr[j-1])
        {
            swap(arr[j], arr[j-1]);
            j--;
        }
    }
}

int main()
{
    int len,temp;
    vector<int> arr;
    cin>>len;
    for(int i=0; i<len; i++)
    {
        cin>>temp;
        arr.push_back(temp);
    }
    insertionSort(arr, len);
    cout<<"sorted:";
    for(int i=0; i<len; i++)
    {
        cout<<arr[i]<<" ";
    }
    return 0;
}